conda-installation.md

```diff
@@ -247,3 +269,71 @@ arguments = myscript.py
 transfer_input_files = Miniconda3-latest-Linux-x86_64.sh, script.py, other_input.file
 ```
 {:.sub}
+
+# Specifying Exact Dependency Versions
+
+An important part of improving reproducibility and consistency between runs
+is to ensure that you use the correct/expected versions of your dependencies.
+
+When you run a command like `conda install numpy`, `conda` tries to install
+**the most recent version of `numpy`**. For example, `numpy` version `1.18.2`
+was released on March 17, 2020. To install exactly this version of `numpy`, you
+would run `conda install numpy==1.18.2` (the same works for `pip`). We
+recommend installing with an explicit version to make sure you have exactly 
+the version of a package that you want. This is often called 
+"pinning" or "locking" the version of the package.
+
+If you want a record of what is installed in your environment, or want to 
+reproduce your environment on another computer, conda can create a file, usually
+called `environment.yml`, that describes the exact versions of all of the 
+packages you have installed in an environment.
+This file can be re-used by a different conda command to recreate that 
+exact environment on another computer.
+
+To create an `environment.yml` file from your currently-activated environment, run
+```
+[alice@submit]$ conda env export > environment.yml
+```
+{: .term}
+
+This `environment.yml` will pin the exact version of *every dependency in your
+environment*. This can sometimes be problematic if you are moving between
+platforms because a package version may not be available on some other platform,
+causing an "unsatisfiable dependency" or "inconsistent environment" error.
+A much less strict pinning is
+```
+[alice@submit]$ conda env export --from-history > environment.yml
+```
+{: .term}
+which only lists packages that you installed manually, and **does not pin their
+versions unless you yourself pinned them during installation**. In general,
+exact environment specifications are not guaranteed to be transfer between
+platforms (e.g., between Windows and CentOS). We recommend using the strictest
```

Comment:
Will "CentOS" be widely recognized as a synonym for "Linux"?

Reply:
I would use replace CentOS with Linux.